src/containers/Authentication/Signin/Signin.tsx

```diff
@@ -15,244 +10,118 @@ import {
-      return;
-    }
-
-    if (!resp.ok) {
-      alert(
-        `Unexpected error ${resp.status} validating your credentials. Please try again later.`
-      );
-      return;
-    }
-
-    let data = await resp.json();
-
-    if (data.accessToken) {
-      await this.props.signin({
-        email: this.state.email,
-        password: this.state.password,
-        accessToken: data.accessToken,
-      });
-      await this.props.getAdvertiser(data);
-    }
-
-    if (data.error) {
-      alert("Error " + data.error);
-      return;
-    }
-
-    if (data.allowCredentials) {
-      this.get(data);
-    }
-
-    this.toggleSubmitting();
-  }
-
-  toggleSubmitting() {
-    this.setState({
-      submitting: !this.state.submitting,
-    });
-  }
-
-  get(attestationObj) {
-    attestationObj.challenge = base64url.toBuffer(attestationObj.challenge);
-
-    attestationObj.allowCredentials.forEach((allowedCredential) => {
-      allowedCredential.id = base64url.toBuffer(allowedCredential.id);
-    });
-
-    navigator.credentials
-      .get({ publicKey: attestationObj })
-      .then(async (cred: any) => {
-        await this.sendClientCredential(cred, attestationObj.userId);
-      })
-      .catch((err) => {
-        console.log("ERROR", err);
-      });
-  }
+import BraveLogo from "../../../..//brave-logotype-full-color.png";
+import { useHistory } from "react-router-dom";
+import { useSignIn } from "../../../auth/hooks/mutations/useSignIn";
 
-  publicKeyCredentialToJSON(pubKeyCred) {
-    if (pubKeyCred instanceof Array) {
-      let arr = [];
-      //@ts-ignore
-      for (let i of pubKeyCred) arr.push(publicKeyCredentialToJSON(i));
-
-      return arr;
-    }
-
-    if (pubKeyCred instanceof ArrayBuffer) {
-      //@ts-ignore
-      return base64url.encode(pubKeyCred);
-    }
-
-    if (pubKeyCred instanceof Object) {
-      let obj = {};
-
-      for (let key in pubKeyCred) {
-        obj[key] = this.publicKeyCredentialToJSON(pubKeyCred[key]);
-      }
-
-      return obj;
-    }
-
-    return pubKeyCred;
-  }
+interface Props {
+  state?: unknown;
+}
 
-  async sendClientCredential(clientCredential, userId) {
-    const resp = await fetch(
-      `${process.env.REACT_APP_SERVER_ADDRESS}/auth/challenge/${userId}`,
-      {
-        method: "POST",
-        mode: "cors",
-        credentials: "same-origin",
-        headers: {
-          "Content-Type": "application/json",
-        },
-        body: JSON.stringify(this.publicKeyCredentialToJSON(clientCredential)),
+export function SignIn({ state }: Props) {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [error, setError] = useState<string>();
+  const history = useHistory();
+
+  const { signIn, loading } = useSignIn({
+    onSuccess() {
+      history.push("/", { triedAuth: true });
+    },
+    onError(msg) {
+      setError(msg);
+    },
+  });
+
+  useEffect(() => {
+    if (state) {
+      const s = state as { triedAuth: boolean };
```

Comment:
Can we avoid the cast here?  Not quite sure why `state` is typed as `unknown`.

Reply:
Verbally discussed: passed in from history.location.state, so needs to be unknown.